extensions/resource-deployment/src/services/azdataService.ts

```diff
@@ -0,0 +1,168 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+import * as cp from 'child_process';
+import * as path from 'path';
+import * as fs from 'fs';
+import { IPlatformService } from './platformService';
+
+export interface DeploymentProfile {
+	name: string;
+	defaultDataSize: string;
+	defaultLogSize: string;
+	master: string;
+	data: string;
+	compute: string;
+	hdfs: string;
+	nameNode: string;
+	spark: string;
+	activeDirectory: boolean;
+	hadr: string;
+	includeSpark: string;
+	gatewayPort: string;
+	appProxyPort: string;
+	sqlServerPort: string;
+	readableSecondaryPort?: string;
+	controllerPort: string;
+	defaultDataStorageClass: string;
+	defaultLogsStorageClass: string;
+	bdcJson: any;
+	controlJson: any;
+}
+
+interface BdcConfigListOutput {
+	stdout: string[];
+}
+
+export interface IAzdataService {
+	getDeploymentProfiles(): Thenable<DeploymentProfile[]>;
+}
+
+const SqlServerMasterResourceName = 'master';
+const DataResourceName = 'data-0';
+const HdfsResourceName = 'storage-0';
+const ComputeResourceName = 'compute-0';
+const NameNodeResourceName = 'nmnode-0';
+const HadrEnabledSettingName = 'hadr.enabled';
+export class AzdataService implements IAzdataService {
+	constructor(private platformService: IPlatformService) {
+	}
+
+	getDeploymentProfiles(): Thenable<DeploymentProfile[]> {
+		process.env['ACCEPT_EULA'] = 'Yes';
+		return this.ensureWorkingDirectoryExists().then(() => {
+			return this.getDeploymentProfileNames();
+		}).then((names: string[]) => {
+			const profilePromises: Thenable<DeploymentProfile>[] = [];
+			names.forEach(name => {
+				profilePromises.push(this.getDeploymentProfileInfo(name));
+			});
+			return Promise.all(profilePromises);
```

Comment:
This can be simplified to

Promise.all(names.map(name => this.getDeploymentProfileInfo(name));